vyper/venom/README.md

```diff
@@ -29,59 +29,39 @@ Venom employs two scopes: global and function level.
 ### Example code
 
 ```llvm
-IRFunction: global
-
-global:
-    %1 = calldataload 0
-    %2 = shr 224, %1
-    jmp label %selector_bucket_0
-
-selector_bucket_0:
-    %3 = xor %2, 1579456981
-    %4 = iszero %3
-    jnz label %1, label %2, %4
-
-1:  IN=[selector_bucket_0] OUT=[9]
-    jmp label %fallback
-
-2:
-    %5 = callvalue
-    %6 = calldatasize
-    %7 = lt %6, 164
-    %8 = or %5, %7
-    %9 = iszero %8
-    assert %9
-    stop
-
-fallback:
-    revert 0, 0
+fn global => {
```

Comment:
it's a little bit weird that functions are curly brace delimited but basic blocks are indent delimited

Reply:
(not a huge deal, but it's something to think about -- either making basic blocks curly brace delimited or functions whitespace delimited)

Reply:
tbh I like the mix, makes it clear which is which, if you really want can change

Reply:
I think I side with @charles-cooper on this. Seems weird to only have braces only for that part. Also is doesn't sit well with me as venom is more of a simple "assembly" like language, so no need for deep nesting.

Reply:
Something like this:

```
fn global:
    %1 = calldataload 0
    %2 = shr 224, %1
    jmp @selector_bucket_0

selector_bucket_0:
    %3 = xor %2, 1579456981
    %4 = iszero %3
    jnz @1, @2, %4

1:
    jmp @fallback

2:
    %5 = callvalue
    %6 = calldatasize
    %7 = lt %6, 164
    %8 = or %5, %7
    %9 = iszero %8
    assert %9
    stop

fallback:
    revert 0, 0

data:
```

Reply:
I don't really like this, makes it hard to differentiate between basic blocks, functions & data.

Reply:
is there consensus on this? I can do it that way, just seems kind of ugly. I know it works in Python because of indents but I don't want to have to deal with indents in the grammar. Down for whatever, just want this to work

Reply:
i don't think there need to be indents in the grammar, since functions are delimited with "function' (which is a reserved keyword)

Reply:
yes but I mean visually